app/src/common/shared/com/igalia/wolvic/VRBrowserActivity.java

```diff
@@ -683,26 +680,44 @@ void loadFromIntent(final Intent intent) {
             }
         }
 
-        Uri uri = intent.getData();
-
         boolean openInWindow = false;
         boolean openInBackground = false;
+        boolean openInKioskMode = false;
+
+        Uri dataUri = intent.getData();
+        Uri targetUri = null;
```

Comment:
Would it be possible to keep the old "uri" name? That way this PR would be simpler, as many changes are just renames

Reply:
We could do that, yes.

I just think that it is clearer to have two different names, so `dataUri` is the one that we get from `intent.getData()` and `targetUri` the one that Wolvic will open (which might not be the same). For example

```java
dataUri = "wolvic://com.igalia.wolvic/?kiosk=true&url=moonrider.xyz"
targetUri = "moonrider.xyz"
```

Reply:
Right I agree with you but I think it's better to do it in a follow up if needed. Just to keep bisects easier.